chessington/engine/pieces.py

```diff
@@ -33,14 +34,90 @@ class Pawn(Piece):
     """
     A class representing a chess pawn.
     """
+
     def get_available_moves(self, board) -> List[Square]:
         current_square = board.find_piece(self)
+        square_list = []
+        if not self.can_move(board, current_square):
+            return square_list
+
+        if self.player == Player.BLACK:
+            square_list.append(Square.at(current_square.row - 1, current_square.col))
+
+            if current_square.row == 6 and not self.has_piece_in_front(board, current_square, distance=2):
+                square_list.append(Square.at(current_square.row - 2, current_square.col))
+        else:
+            square_list.append(Square.at(current_square.row + 1, current_square.col))
+            if current_square.row == 1 and not self.has_piece_in_front(board, current_square, distance=2):
+                square_list.append(Square.at(current_square.row + 2, current_square.col))
+        square_list += self.get_moves_diag(board, current_square)
+
+        return square_list
+
+    def has_piece_in_front(self, board: Board, current_square, distance=1) -> bool:
+        if self.player == Player.BLACK:
+            if board.get_piece(Square.at(current_square.row - distance, current_square.col)):
+                return True
+        else:
+            if board.get_piece(Square.at(current_square.row + distance, current_square.col)):
+                return True
```

Comment:
Same commonisation tactic here: 

- Check if the piece is white or black and multiply the distance by `-1` - that should spare us an `if` branch.